### ISSUE TITLE
No routing around highway=pedestrian areas

### ISSUE THREAD

**Issue by FranckRobin** [valhalla/valhalla#3736]
I understand from previous issues that there is no plan for pedestrian to route via highway=platform (#3691), or through areas (#17), but it would be very useful to route at least around highway=pedestrian areas (following their outer)?
I did it successfully with OSRM and GraphHopper (in https://www.openstreetmap.org), but not with valhalla and mapbox whatever the settings (including default ones).
This leads to very incorrect paths in cases similar as the one below: 16km instead of 16m
https://valhalla.openstreetmap.de/directions?profile=pedestrian&wps=5.374277830123902%2C43.29464496683285%2C5.374014973640443%2C43.29465668009512
![image](https://user-images.githubusercontent.com/112408722/189351861-55ea2703-af8b-4761-baa1-1385e83a669a.png)
Could you please suggest me a workaround, or a solution? Thank you by advance

**Comment by kevinkreiser** [comment]
seems like a reasonable feature request at the very least to achieve parity with other routers. to me, although i know routing on outers or inners of an area is technically wrong there is a point to be made about still allowing routing when there are no other options currently. in other words there is an argument that goes like: "its better than nothing". @FranckRobin do you have time to make a quick test? i think you might only need to remove this little snippet of code to get the result you are looking for: https://github.com/valhalla/valhalla/blob/master/lua/graph.lua#L1399-L1402

basically delete or comment those lines out. compile valhalla. build the tiles, and check your route to see if it does what you want.

**Comment by FranckRobin** [comment]
I would be happy to do this test , but it probably would't be fast as I never built valhalla, just installed it with docker-compose, and used it.
Looking at your suggestion and this snippet, would't it be too impacting to stop excluding all area=yes? Should't we just add other conditions to this code, such as `if kv["area"] == "yes" and kv["highway"] ~= "pedestrian"`? Thanks

**Comment by kevinkreiser** [comment]
@FranckRobin yeah it would affect all areas that were marked with highway tags not just highway tags with pedestrian values. for the purposes of just testing it out i think its fine to not be so surgical but you're right, if we wanted to actually ship it we might want to narrow the scope to pedestrian or another subset of travel modes.

**Comment by osmuser63783** [comment]
It would be nice to see this fixed. Valhalla refusing to route through or along the edges of pedestrian areas leads to some strange results:
https://www.openstreetmap.org/directions?engine=fossgis_valhalla_foot&route=55.95494%2C-3.19009%3B55.95429%2C-3.18904#map=17/55.95440/-3.19020&layers=N
